### PR TITLE
Avoid casting on encode and decode enums (#167)

### PR DESCRIPTION
**Pull request by cguimaraes**
Fixes https://github.com/eclipse-zenoh/zenoh-pico/issues/167.

### CODE REVIEW

**Comment by cguimaraes** [comment]
Fixes #167.

File: src/protocol/codec.c [hunk 1]
```diff
@@ -49,18 +49,50 @@ int8_t _z_period_decode_na(_z_period_t *p, _z_zbuf_t *buf) {
...
-    *en = 0;
 
-    if (_z_zbuf_can_read(zbf) == true) {
-        *en = _z_zbuf_read(zbf);
-    } else {
-        _Z_DEBUG("WARNING: Not enough bytes to read\n");
-        ret |= _Z_ERR_MESSAGE_DESERIALIZATION_FAILED;
-    }
+    _z_zint_t tmp;
+    _z_zint_decode(&tmp, zbf);
```
Comment (Mallets) [inline src/protocol/codec.c:58]: This should be `ret |= _z_zint_decode(&tmp, zbf);`

File: src/protocol/codec.c [hunk 2]
```diff
@@ -49,18 +49,50 @@ int8_t _z_period_decode_na(_z_period_t *p, _z_zbuf_t *buf) {
...
+    return ret;
+}
+
+int8_t _z_consolidation_mode_encode(_z_wbuf_t *wbf, z_consolidation_mode_t en) { return _z_zint_encode(wbf, en); }
+
+int8_t _z_consolidation_mode_decode(z_consolidation_mode_t *en, _z_zbuf_t *zbf) {
+    int8_t ret = _Z_RES_OK;
+
+    _z_zint_t tmp;
+    _z_zint_decode(&tmp, zbf);
```
Comment (Mallets) [inline src/protocol/codec.c:70]: This should be `ret |= _z_zint_decode(&tmp, zbf);`

File: src/protocol/codec.c [hunk 3]
```diff
@@ -49,18 +49,50 @@ int8_t _z_period_decode_na(_z_period_t *p, _z_zbuf_t *buf) {
...
+    return ret;
+}
+
+int8_t _z_query_target_encode(_z_wbuf_t *wbf, z_query_target_t en) { return _z_zint_encode(wbf, en); }
+
+int8_t _z_query_target_decode(z_query_target_t *en, _z_zbuf_t *zbf) {
+    int8_t ret = _Z_RES_OK;
+
+    _z_zint_t tmp;
+    _z_zint_decode(&tmp, zbf);
```
Comment (Mallets) [inline src/protocol/codec.c:82]: This should be `ret |= _z_zint_decode(&tmp, zbf);`

File: src/protocol/codec.c [hunk 4]
```diff
@@ -49,18 +49,50 @@ int8_t _z_period_decode_na(_z_period_t *p, _z_zbuf_t *buf) {
...
+    return ret;
+}
+
+int8_t _z_whatami_encode(_z_wbuf_t *wbf, z_whatami_t en) { return _z_zint_encode(wbf, en); }
+
+int8_t _z_whatami_decode(z_whatami_t *en, _z_zbuf_t *zbf) {
+    int8_t ret = _Z_RES_OK;
+
+    _z_zint_t tmp;
+    _z_zint_decode(&tmp, zbf);
```
Comment (Mallets) [inline src/protocol/codec.c:94]: This should be `ret |= _z_zint_decode(&tmp, zbf);`

Comment (cguimaraes) [reply src/protocol/codec.c]: fixed.

File: src/protocol/codec.c (hunk 3 above)
Comment (cguimaraes) [reply src/protocol/codec.c]: fixed.

File: src/protocol/codec.c (hunk 2 above)
Comment (cguimaraes) [reply src/protocol/codec.c]: fixed.

File: src/protocol/codec.c (hunk 1 above)
Comment (cguimaraes) [reply src/protocol/codec.c]: fixed.

**Review by Mallets — APPROVED** [review]

**Pull request merged**